tests/Microsoft.Identity.Test.Unit.net45/CoreTests/WsTrustTests/WsTrustEndpointTests.cs

```diff
@@ -164,6 +165,39 @@ public void TestWsTrustEndpointUsernamePasswordWsTrust13()
             Assert.AreEqual(expectedMessage, requestMessage);
         }
 
+        [TestMethod]
+        [Description("Assert client info is included in userrealm discovery header")]
+        public void UserrealmDiscoveryHeadersTest()
+        {
+            string endpoint = "https://login.microsoftonline.com/common/userrealm/";
+
+            using (var harness = new MockHttpAndServiceBundle())
```

Comment:
Do you need all this harness stuff if you just want to unit test CreateUserrealmEndpoint which seems to need 2 strings and a logger?

Either simplify the test, or make the method CreateUserrealmEndpoint  private and get the Uri from the _httpManager mock (I'd go for the second approach because we can keep CreateUserrealmEndpoint   nice and private)

Reply:
i tried to simplify this, but maybe we can sync in the morning. i have a few questions for you.

---
In reply to: [251527049](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/840#discussion_r251527049) [](ancestors = 251527049)